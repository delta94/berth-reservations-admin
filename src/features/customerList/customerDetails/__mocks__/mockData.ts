import { CustomerListApplication, CustomerListBerthLeases, CustomerListBill, CustomerListBoat } from '../../types';
import { CustomerGroup } from '../../../../@types/__generated__/globalTypes';
import { CustomerDetailsProps } from '../CustomerDetails';

export const customerListEntry: Omit<
  CustomerDetailsProps,
  'berths' | 'winterStoragePlaces' | 'boats' | 'applications' | 'bills'
> = {
  name: 'Mikko Matias Mallikas',
  address: 'Telakkakatu 1 A 10',
  postalCode: '00100',
  city: 'Helsinki',
  phone: '+358 040 123 4567',
  email: 'mikko.mallikas@meri.fi',
  customerGroup: CustomerGroup.PRIVATE,
  comment:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean malesuada id est ut pellentesque. ' +
    'Vivamus quis maximus sem. Ut auctor vestibulum mattis. Vestibulum mollis diam convallis ligula consequat ' +
    'sagittis.',
};

export const customerListWinterStoragePlaces = [{ id: '123', title: 'Rajasaari' }];

export const customerListBoats: CustomerListBoat[] = [{ id: '123', name: 'Cama la Yano' }];

export const customerListApplications: CustomerListApplication[] = [{ id: '123', createdAt: '2019-12-01' }];

export const customerListBills: CustomerListBill[] = [{ id: '123', date: '2020-01-21' }];

export const customerListBerthLeases: CustomerListBerthLeases[] = [
  { id: '123', title: 'Pursilahdenranta B31' },
  { id: '321', title: 'Str√∂msinlahdenranta B31' },
];
