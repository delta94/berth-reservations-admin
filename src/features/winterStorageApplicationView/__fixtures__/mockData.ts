import {
  INDIVIDUAL_WINTER_STORAGE_APPLICATION,
  INDIVIDUAL_WINTER_STORAGE_APPLICATION_boatTypes as BOAT_TYPES,
  INDIVIDUAL_WINTER_STORAGE_APPLICATION_winterStorageApplication as WINTER_STORAGE_APPLICATION,
} from '../__generated__/INDIVIDUAL_WINTER_STORAGE_APPLICATION';
import { ApplicationStatus, BerthApplicationLanguage } from '../../../@types/__generated__/globalTypes';

export const mockData: INDIVIDUAL_WINTER_STORAGE_APPLICATION & {
  winterStorageApplication: WINTER_STORAGE_APPLICATION;
  boatTypes: BOAT_TYPES[];
} = {
  winterStorageApplication: {
    id: 'V2ludGVyU3RvcmFnZUFwcGxpY2F0aW9uTm9kZTox',
    firstName: 'test',
    lastName: 'test',
    address: 'test',
    municipality: 'helsinki',
    zipCode: '00100',
    phoneNumber: '555555',
    email: 'test@test.com',
    businessId: '',
    companyName: '',
    language: BerthApplicationLanguage.EN,
    customer: null,
    createdAt: '2019-05-06T10:54:21.746387+00:00',
    boatType: '3',
    boatRegistrationNumber: 'test',
    boatWidth: 1,
    boatLength: 1,
    boatName: 'test',
    boatModel: 'test',
    acceptBoatingNewsletter: true,
    acceptFitnessNews: true,
    acceptLibraryNews: true,
    acceptOtherCultureNews: true,
    status: ApplicationStatus.PENDING,
    winterStorageAreaChoices: [
      {
        winterStorageArea: null,
        priority: 1,
        winterStorageAreaName: 'Iso-Sarvasto',
        __typename: 'WinterStorageAreaChoiceType',
      },
      {
        winterStorageArea: null,
        priority: 2,
        winterStorageAreaName: 'Rajasaari',
        __typename: 'WinterStorageAreaChoiceType',
      },
    ],
    applicationCode: '1234',
    __typename: 'WinterStorageApplicationNode',
  },
  boatTypes: [
    {
      id: '1',
      name: 'Jollavene',
      __typename: 'BoatTypeType',
    },
    {
      id: '2',
      name: 'Soutuvene',
      __typename: 'BoatTypeType',
    },
    {
      id: '3',
      name: 'Perämoottorivene',
      __typename: 'BoatTypeType',
    },
    {
      id: '4',
      name: 'Sisäperämoottorivene',
      __typename: 'BoatTypeType',
    },
    {
      id: '5',
      name: 'Keskimoottorivene',
      __typename: 'BoatTypeType',
    },
    {
      id: '6',
      name: 'Purjevene / moottoripursi',
      __typename: 'BoatTypeType',
    },
    {
      id: '7',
      name: 'Troolari',
      __typename: 'BoatTypeType',
    },
    {
      id: '8',
      name: 'Suuri alus (yli 20t)',
      __typename: 'BoatTypeType',
    },
  ],
};
