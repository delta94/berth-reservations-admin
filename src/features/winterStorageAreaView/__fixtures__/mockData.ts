import { INDIVIDUAL_WINTER_STORAGE_AREA } from '../__generated__/INDIVIDUAL_WINTER_STORAGE_AREA';
import { LeaseStatus } from '../../../@types/__generated__/globalTypes';

export const mockData: INDIVIDUAL_WINTER_STORAGE_AREA = {
  winterStorageArea: {
    properties: {
      name: 'WS Area 1',
      zipCode: '00100',
      municipality: 'Helsinki',
      streetAddress: '',
      wwwUrl: '',
      imageFile: null,
      maps: [],
      sections: {
        edges: [
          {
            node: {
              id: 'V2ludGVyU3RvcmFnZVNlY3Rpb25Ob2RlOjI1OTgyMTEyLTllNDktNGUwNy1hYmVjLTZhMTQyMzcyZmZjOQ==',
              properties: {
                identifier: '-',
                electricity: true,
                water: true,
                gate: false,
                summerStorageForBoats: false,
                summerStorageForTrailers: false,
                summerStorageForDockingEquipment: false,
                leases: {
                  __typename: 'WinterStorageLeaseNodeConnection',
                  edges: [],
                },
                places: {
                  edges: [
                    {
                      node: {
                        id: 'V2ludGVyU3RvcmFnZVBsYWNlTm9kZTo0YzUzZTY3NC0zMmFjLTRkNTEtYmQzMC0wNTcyMWNjMjUyMTE=',
                        number: 1,
                        width: 3,
                        length: 5,
                        isActive: true,
                        leases: {
                          edges: [
                            {
                              node: {
                                id: 'V2ludGVyU3RvcmFnZUxlYXNlTm9kZTo3ZTU0NjdmMy1jZGJhLTQzMmEtODdiZi05ODBiZTI1ZGFjNzg=',
                                startDate: '2020-09-15',
                                endDate: '2021-06-10',
                                status: LeaseStatus.DRAFTED,
                                isActive: false,
                                application: {
                                  createdAt: '2020-07-17T10:52:57.079036+00:00',
                                  customer: {
                                    id: 'UHJvZmlsZU5vZGU6NWNjYzgyNDUtNmRiOS00YTRiLWI5NTEtNWYxNDQ5YTY5NzY2',
                                    firstName: 'Heikki',
                                    lastName: 'Kinnunen',
                                    __typename: 'ProfileNode',
                                  },
                                  id: 'foo',
                                  __typename: 'WinterStorageApplicationNode',
                                },
                                __typename: 'WinterStorageLeaseNode',
                              },
                              __typename: 'WinterStorageLeaseNodeEdge',
                            },
                          ],
                          __typename: 'WinterStorageLeaseNodeConnection',
                        },
                        __typename: 'WinterStoragePlaceNode',
                      },
                      __typename: 'WinterStoragePlaceNodeEdge',
                    },
                    {
                      node: {
                        id: 'V2ludGVyU3RvcmFnZVBsYWNlTm9kZTo3OTkzYzAxOC0zMTMwLTRjNTItYjA3Ni1mMGZlN2Q5YjRmYTE=',
                        number: 2,
                        width: 3,
                        length: 5,
                        isActive: true,
                        leases: {
                          edges: [
                            {
                              node: {
                                id: 'V2ludGVyU3RvcmFnZUxlYXNlTm9kZTpjOWNjODAyNi1hZTkzLTRjMTUtYmJiNy1kM2Q1ZjE4ZjViYjA=',
                                startDate: '2020-09-15',
                                endDate: '2021-06-10',
                                status: LeaseStatus.DRAFTED,
                                isActive: false,
                                application: null,
                                __typename: 'WinterStorageLeaseNode',
                              },
                              __typename: 'WinterStorageLeaseNodeEdge',
                            },
                          ],
                          __typename: 'WinterStorageLeaseNodeConnection',
                        },
                        __typename: 'WinterStoragePlaceNode',
                      },
                      __typename: 'WinterStoragePlaceNodeEdge',
                    },
                  ],
                  __typename: 'WinterStoragePlaceNodeConnection',
                },
                __typename: 'WinterStorageSectionProperties',
              },
              __typename: 'WinterStorageSectionNode',
            },
            __typename: 'WinterStorageSectionNodeEdge',
          },
        ],
        __typename: 'WinterStorageSectionNodeConnection',
      },
      __typename: 'WinterStorageAreaProperties',
    },
    __typename: 'WinterStorageAreaNode',
  },
};
