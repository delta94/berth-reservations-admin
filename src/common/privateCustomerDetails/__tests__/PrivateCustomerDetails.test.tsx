import React from 'react';
import { shallow } from 'enzyme';
import { HashRouter } from 'react-router-dom';

import PrivateCustomerDetails, { PrivateCustomerDetailsProps } from '../PrivateCustomerDetails';
import { privateCustomerProfile as mockData } from '../__fixtures__/mockData';

const mockProps: PrivateCustomerDetailsProps = {
  firstName: mockData.firstName,
  lastName: mockData.lastName,
  language: mockData.language,
};

describe('PrivateCustomerDetails', () => {
  const getWrapper = (props = {}) =>
    shallow(
      <HashRouter>
        <PrivateCustomerDetails {...mockProps} {...props} />
      </HashRouter>
    );

  it('renders normally with minimum props', () => {
    const wrapper = getWrapper();

    expect(wrapper.render()).toMatchSnapshot();
  });

  it('renders normally with all props', () => {
    const wrapper = getWrapper({
      ...mockData,
      title: 'Test',
      showCustomerNameAsLink: true,
    });

    expect(wrapper.render()).toMatchSnapshot();
  });
});
