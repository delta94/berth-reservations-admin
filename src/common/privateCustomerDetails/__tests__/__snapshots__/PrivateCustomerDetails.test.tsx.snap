// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateCustomerDetails renders normally with all props 1`] = `
Array [
  <article
    class="section"
  >
    <h4
      class="text standard h4 m title"
    >
      Test
    </h4>
    <section
      class="body"
    >
      <div
        class="labelValuePair"
      >
        <span
          class="label standard"
        >
          Etunimet:
        </span>
        <span
          class="value left"
        >
          <a
            class="internalLink brand"
            href="#/customers/0"
            title="Testi"
          >
            Testi
          </a>
        </span>
      </div>
      <div
        class="labelValuePair"
      >
        <span
          class="label standard"
        >
          Sukunimi:
        </span>
        <span
          class="value left"
        >
          <a
            class="internalLink brand"
            href="#/customers/0"
            title="Käyttäjä"
          >
            Käyttäjä
          </a>
        </span>
      </div>
      <div
        class="labelValuePair"
      >
        <span
          class="label standard"
        >
          Henkilötunnus:
        </span>
        <span
          class="value left"
        >
          <a
            class="internalLink brand"
            href="#/customers/0"
            title="010101A1234"
          >
            010101A1234
          </a>
        </span>
      </div>
    </section>
  </article>,
  <article
    class="section"
  >
    <section
      class="body"
    >
      <div
        class="labelValuePair"
      >
        <span
          class="label standard"
        >
          Jakeluosoite:
        </span>
        <span
          class="value left"
        >
          Testikatu 1
        </span>
      </div>
      <div
        class="labelValuePair"
      >
        <span
          class="label standard"
        >
          Postinumero:
        </span>
        <span
          class="value left"
        >
          00100
        </span>
      </div>
      <div
        class="labelValuePair"
      >
        <span
          class="label standard"
        >
          Postitoimipaikka:
        </span>
        <span
          class="value left"
        >
          Helsinki
        </span>
      </div>
    </section>
  </article>,
  <article
    class="section"
  >
    <section
      class="body"
    >
      <div
        class="labelValuePair"
      >
        <span
          class="label standard"
        >
          Matkapuhelin:
        </span>
        <span
          class="value left"
        >
          0504391742
        </span>
      </div>
      <div
        class="labelValuePair"
      >
        <span
          class="label standard"
        >
          Sähköpostiosoite:
        </span>
        <span
          class="value left"
        >
          test@example.com
        </span>
      </div>
    </section>
  </article>,
  <article
    class="section"
  >
    <section
      class="body"
    >
      <div
        class="labelValuePair"
      >
        <span
          class="label standard"
        >
          Asiakasryhmä:
        </span>
        <span
          class="value left"
        >
          Yksityinen
        </span>
      </div>
      <div
        class="labelValuePair"
      >
        <span
          class="label standard"
        >
          Asiointikieli:
        </span>
        <span
          class="value left"
        >
          Suomi
        </span>
      </div>
      <div
        class="labelValuePair"
      >
        <span
          class="label standard"
        />
        <span
          class="value left"
        >
          Laskutus paperisena
        </span>
      </div>
    </section>
  </article>,
  <article
    class="section"
  >
    <section
      class="body"
    >
      <div
        class="labelValuePair"
      >
        <span
          class="label standard"
        >
          Huomiot:
        </span>
        <span
          class="value left"
        >
          Testikäyttäjä
        </span>
      </div>
    </section>
  </article>,
]
`;

exports[`PrivateCustomerDetails renders normally with minimum props 1`] = `
Array [
  <article
    class="section"
  >
    <h4
      class="text standard h4 m title"
    >
      HENKILÖTIEDOT
    </h4>
    <section
      class="body"
    >
      <div
        class="labelValuePair"
      >
        <span
          class="label standard"
        >
          Etunimet:
        </span>
        <span
          class="value left"
        >
          Testi
        </span>
      </div>
      <div
        class="labelValuePair"
      >
        <span
          class="label standard"
        >
          Sukunimi:
        </span>
        <span
          class="value left"
        >
          Käyttäjä
        </span>
      </div>
    </section>
  </article>,
  <article
    class="section"
  >
    <section
      class="body"
    >
      <div
        class="labelValuePair"
      >
        <span
          class="label standard"
        >
          Jakeluosoite:
        </span>
        <span
          class="value left"
        >
          -
        </span>
      </div>
      <div
        class="labelValuePair"
      >
        <span
          class="label standard"
        >
          Postinumero:
        </span>
        <span
          class="value left"
        >
          -
        </span>
      </div>
      <div
        class="labelValuePair"
      >
        <span
          class="label standard"
        >
          Postitoimipaikka:
        </span>
        <span
          class="value left"
        >
          -
        </span>
      </div>
    </section>
  </article>,
  <article
    class="section"
  >
    <section
      class="body"
    >
      <div
        class="labelValuePair"
      >
        <span
          class="label standard"
        >
          Matkapuhelin:
        </span>
        <span
          class="value left"
        >
          -
        </span>
      </div>
      <div
        class="labelValuePair"
      >
        <span
          class="label standard"
        >
          Sähköpostiosoite:
        </span>
        <span
          class="value left"
        >
          -
        </span>
      </div>
    </section>
  </article>,
  <article
    class="section"
  >
    <section
      class="body"
    >
      <div
        class="labelValuePair"
      >
        <span
          class="label standard"
        >
          Asiakasryhmä:
        </span>
        <span
          class="value left"
        >
          Yksityinen
        </span>
      </div>
      <div
        class="labelValuePair"
      >
        <span
          class="label standard"
        >
          Asiointikieli:
        </span>
        <span
          class="value left"
        >
          Suomi
        </span>
      </div>
    </section>
  </article>,
  <article
    class="section"
  >
    <section
      class="body"
    >
      <div
        class="labelValuePair"
      >
        <span
          class="label standard"
        >
          Huomiot:
        </span>
        <span
          class="value left"
        >
          -
        </span>
      </div>
    </section>
  </article>,
]
`;
